### PR TITLE
fix the  vue init error from private repository  in window OS

### PR DESCRIPTION
**Pull request by kospot**
if the file name has symbol ':', will make this error like :
'git clone' failed with status 128

### CODE REVIEW

File: bin/vue-init [hunk 1]
```diff
@@ -65,7 +65,7 @@ var name = inPlace ? path.relative('../', process.cwd()) : rawName
 var to = path.resolve(rawName || '.')
 var clone = program.clone || false
 
-var tmp = path.join(home, '.vue-templates', template.replace(/\//g, '-'))
+var tmp = path.join(home, '.vue-templates', template.replace(/[\/|:]/g, '-'))
```
Comment (posva) [inline bin/vue-init]: when do you need to replace the `|` character?

Comment (kospot) [inline bin/vue-init]: the | character means or

and is that you planing publish the npm package ?

Comment (posva) [inline bin/vue-init]: using `[]` will already do an _or_
I don't get your question, can you rephrase it, please?

**Comment by rwb7041** [comment]
Is there a timeline of when this change will be added to the cli and published to NPM? This change will help me out a lot as I have a custom template I've been using out of a private repo and I can use that template on my mac just fine, but I cant on my windows machine due to the filename error.

**Comment by weizhenye** [comment]
@posva @yyx990803 

Could please you modify this PR yourself and merge it? There are hundreds issues and PRs ( #438 #518 #525 #567 ) trying to fix this bug which already exists 8 months. Or just reopen and merge #567

**Comment by posva** [comment]
This PR is not solving those issues... It's still not clear to me what is this solving that doesn't work on current vue-cli version

**Comment by rwb7041** [comment]
@posva It's solving a windows error that prevents the vue-cli from being able to use template that are not on github, such as templates in private bitbucket repositories. The cli fails because it tries to make a folder with a colon in the name, and colons are not allowed in file/folder names on windows, just like the \ character. (I found the full list of reserved characters for Windows here: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx)
